src/lightning/pytorch/callbacks/memory_format.py

```diff
@@ -0,0 +1,80 @@
+# Copyright The Lightning AI team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+MemoryFormat
+===============
+
+changes the model memory format
+"""
+
+from collections.abc import MutableSequence
+from typing import Any, Optional
+
+import torch
+
+import lightning.pytorch as pl
+from lightning.pytorch.callbacks import Callback
+from lightning.pytorch.utilities.rank_zero import rank_zero_warn
+
+
+class MemoryFormat(Callback):
```

Comment:
@Pedrexus, just thinking if the name is right, since it rather does `ReorderImageDimensions`